storage/innobase/xtrabackup/test/run.sh

```diff
@@ -283,6 +283,8 @@ function set_vars()
     then
         PATH="$PWD/..:$PWD/../src:$PWD/../src/xbcloud:$PATH"
         XTRABACKUP_BASEDIR="$PWD/../src"
+    else
+      XTRABACKUP_BASEDIR="$PWD/../"
```

Comment:
What does this allow us to do?

Reply:
`XTRABACKUP_BASEDIR` is used by keyring tests. Jenkins run this on an in-source build, however, if you use this on a folder of result from `make install` SRC folder is not present. This change sets `XTRABACKUP_BASEDIR` on those cases.

Also:
storage/innobase/xtrabackup/test/inc/keyring_file.sh
storage/innobase/xtrabackup/test/inc/keyring_vault.sh
```
if test -d ${XTRABACKUP_BASEDIR}/keyring
then
  plugin_dir=${XTRABACKUP_BASEDIR}/keyring
else
  plugin_dir=${XTRABACKUP_BASEDIR}/lib/plugin
fi
```
Added the same validation on test cases to use either keyring folder(in-source build) or lib/plugin